README.md


<div id="badges"><!-- pkgdown markup -->
<a href="https://CRAN.R-project.org/web/checks/check_results_aroma.cn.html"><img border="0" src="https://www.r-pkg.org/badges/version/aroma.cn" alt="CRAN check status"/></a> <a href="https://github.com/HenrikBengtsson/aroma.cn/actions?query=workflow%3AR-CMD-check"><img border="0" src="https://github.com/HenrikBengtsson/aroma.cn/actions/workflows/R-CMD-check.yaml/badge.svg?branch=develop" alt="R CMD check status"/></a>     <a href="https://app.codecov.io/gh/HenrikBengtsson/aroma.cn"><img border="0" src="https://codecov.io/gh/HenrikBengtsson/aroma.cn/branch/develop/graph/badge.svg" alt="Coverage Status"/></a> 
</div>

# aroma.cn: Copy-Number Analysis of Large Microarray Data Sets 


## Installation
R package aroma.cn is available on [CRAN](https://cran.r-project.org/package=aroma.cn) and can be installed in R as:
```r
install.packages("aroma.cn")
```


### Pre-release version

To install the pre-release version that is available in Git branch `develop` on GitHub, use:
```r
remotes::install_github("HenrikBengtsson/aroma.cn", ref="develop")
```
This will install the package from source.  

<!-- pkgdown-drop-below -->


## Contributing

To contribute to this package, please see [CONTRIBUTING.md](CONTRIBUTING.md).



NEWS
Package: aroma.cn
=================

Version: 1.6.1-9001 [2022-03-09]

 * drawC1C2Density() gained argument 'grid'.
 
 * Utilizing subsetted calculations of matrixStats (>= 0.50.0).
 
 * CLEANUP: Now importing generic extractC1C2() from PSCBS to
  avoid creating a new one.
  
 * CLEANUP: Deprecated nbrOfFiles() methods; please use length()
   on corresponding input data sets.
   
 * CLEANUP: Removed defunct callPeaks() for data.frame:s.


Version: 1.6.1 [2015-10-27]

 * Package now requires R (>= 3.1.1) released July 2014. This allows
   us to use BioC (>= 3.0) (October 2014).
   
 * Bumped package dependencies.
 
 * ROBUSTNESS: Explicitly importing core R functions.
 
 * CLEANUP: Removed makeSmoothSplinePredict() defunct since Aug 2013.
   Made callPeaks() for data.frame:s defunct (was deprecated).


Version: 1.6.0 [2015-02-23]

 * ROBUSTNESS: Added the first package tests.
 
 * Bumped package dependencies.
 
 * Package now requires R (>= 3.0.3) and Bioc (>= 2.13) which were
   released March 2014 and are in fact old; it's recommended to use
   a more recent version of R.


Version: 1.5.9 [2015-01-06]

 * ROBUSTNESS: Package now does a better job importing objects from
   suggested packages.


Version: 1.5.8 [2014-09-04]

 * ROBUSTNESS: It could be that process() for AbstractCurveNormalization
   would generate an error due to read-only permissions introduced
   by copying the target file without resetting the file permissions.
   
 * Added a few missing NAMESPACE imports.


Version: 1.5.7 [2014-06-14]

 * Package now requires R (>= 3.0.0) and BioC (>= 2.13), which were
   released April 2013 and are in fact old and it's recommended to
   use a more recent version of R.
   
 * Updated package dependencies.


Version: 1.5.6 [2014-03-31]

 * Added doSegmentByPairedPSCBS() for AromaUnitPscnBinarySet.


Version: 1.5.5 [2014-03-09]

 * Updated package dependencies.
 
 * Package requires R (>= 2.15.1) [and Bioc (>= 2.11.0)].


Version: 1.5.4 [2014-02-03]

 * ROBUSTNESS: Now points() for C1C2 passes (modified) argument 'x' to
   NextMethod() as 'object=x'.


Version: 1.5.3 [2014-01-30]

 * Updated package dependencies.


Version: 1.5.2 [2013-12-17]

 * CLEANUP: Package no longer uses ':::' in calls.


Version: 1.5.1 [2013-10-23]

 * CLEANUP: Removed several internal prototype methods no longer
   needed or that have been moved to the Mikado package.


Version: 1.5.0 [2013-10-17]

 * Minor tweaks to NAMESPACE.
 
 * Updated package dependencies.
 
 * Package requires R (>= 2.15.0) [and Bioc (>= 2.10.0)].


Version: 1.4.6 [2013-10-07]

 * ROBUSTNESS: Now importing only what needs to be imported and
   formally declaring all S3 methods in NAMESPACE.
   
 * CLEANUP: Dropped obsolete usage of autoload().


Version: 1.4.5 [2013-09-28]

 * Now the 'aroma.cn' Package object is also available when the
   package is only loaded (but not attached).
   
 * Updated package dependencies.


Version: 1.4.4 [2013-09-26]

 * BUG FIX: Forgot to import several functions from 'matrixStats'.
   These went undetected because 'aroma.light' (< 1.31.6) attached
   the 'matrixStats' in the past.


Version: 1.4.3 [2013-09-20]

 * CLEANUP: Now importing only what is needed from PSCBS.
 
 * Updated package dependencies.


Version: 1.4.2 [2013-08-21]

 * More internal updates.


Version: 1.4.1 [2013-08-12]

 * BUG FIX: byPath(), byName() and findByPath() for PairedPSCBSFileSet
   was also affected by the bug described in the R-devel thread
   'Do *not* pass '...' to NextMethod() - it'll do it for you; missing
   documentation, a bug or just me?' on Oct 16, 2012.
   
 * BUG FIX: getPath() for PairedPscbsModel would throw an error on
   getInputDataSet() not defined.


Version: 1.4.0 [2013-08-04]

 * SPEEDUP: Replaced all rm() calls with NULL assignments.
   Also removed several explicit garbage collector calls.
   
 * CLEANUP: The formal package dependency on Bioconductor package
   'aroma.light' has been relaxed so the package can be installed
   without it.
   
 * CLEANUP: Package now only imports R.oo.
 
 * CLEANUP: Defuncted makeSmoothSplinePredict().
 
 * Updated package dependencies.


Version: 1.3.4 [2013-05-20]

 * CRAN POLICY: Now all Rd \usage{} lines are at most 90 characters long.
 
 * CRAN POLICY: Now all Rd example lines are at most 100 characters long.


Version: 1.3.3 [2013-04-22]

 * CLEANUP: findNeutralCopyNumberState() is now in PSCBS.
 
 * Utilizing new startupMessage() of R.oo.


Version: 1.3.2 [2013-04-22]

 * Updated package dependencies.
 
 * CLEANUP: No longer using deprecated PSCBS methods.
 
 * ROBUSTNESS: (load|save)Cache() from R.cache are now explicitly
   imported in the namespace.


Version: 1.3.1 [2013-01-17]

 * Updated internal methods for PairedPSCBS to recognize when other
   mean-level estimators than the sample mean have been used.


Version: 1.3.0 [2013-01-07]

 * Bumped up the package dependencies.
 
 * BUG FIX: process() for PairedPscbsCaller used the global 'verbose'.


Version: 1.2.20 [2012-12-19]

 * BUG FIX: Some example() scripts used non-defined values.


Version: 1.2.19 [2012-11-26]

 * Bumped up the package dependencies.


Version: 1.2.18 [2012-11-21]

 * Now applicable classes utilize the new ParametersInterface.
 
 * DOCUMENTATION: Hiding more internal methods from the help indices.


Version: 1.2.17 [2012-11-13]

 * CLEANUP/FIX: Used "cache:" field modified instead of "cached:".
   After correction, all clearCache() methods could be dropped.


Version: 1.2.16 [2012-11-12]

 * CLEANUP: Now seq_along(x) instead of seq(along=x) everywhere.
   Similarly, seq(ds) where 'ds' is GenericDataFileSet is now
   replaced by seq_along(ds).  Likewise, seq_len(x) replaces
   seq(length=x), and length(ds) replaces nbrOfFiles(ds).


Version: 1.2.15 [2012-11-05]

 * CLEANUP: Replaced all whichVector() with which(), because the
   latter is now the fastest again.


Version: 1.2.14 [2012-11-01]

 * ROBUSTNESS: Now package also imports PSCBS to please 'R CMD check'.

R/C1C2.R
setMethodS3("extractC1C2", "list", function(fitList, ...) {
  c1c2List <- lapply(fitList, FUN=function(fit) {
    extractC1C2(fit, ...)
  })

  # Append NAs between chromosomes
  c1c2TList <- lapply(c1c2List, FUN=function(c1c2) {
    rbind(c1c2, NA)
  })

  c1c2 <- Reduce(rbind, c1c2TList)

  w <- sqrt(c1c2[,4])
  w <- w / sum(w, na.rm=TRUE)
  w <- w / mean(w, na.rm=TRUE)
  c1c2 <- cbind(c1c2, w=w)

  class(c1c2) <- unique(c("C1C2", class(c1c2)))

  c1c2
})

setMethodS3("plot", "C1C2", function(x, xlim=c(0,4), ylim=xlim, xlab=expression(C[1]), ylab=expression(C[2]), ...) {
  plot(NA, xlab=xlab, ylab=ylab, xlim=xlim, ylim=ylim)
  abline(a=0, b=1, lty=3)
  grid()
  points(x, ...)
})

setMethodS3("points", "C1C2", function(x, cex=sqrt(x[,"w"])+1/8, ...) {
  x <- x[,c("C1","C2"),drop=FALSE]
  NextMethod("points", object=x, cex=cex)
})


setMethodS3("fitLoess2D", "C1C2", function(X, Y, ...) {
  fit <- fitLoessKD(X=X[,1:2,drop=FALSE], Y=Y[,1:2,drop=FALSE])
  class(fit) <- c("Loess2DFit", class(fit))
  fit
})

setMethodS3("normalizeLoess2D", "C1C2", function(X, ...) {
  XN <- X
  XN[,1:2] <- normalizeLoessKD(X[,1:2], ...)
  XN
}) # normalizeLoess2D()


##############################################################################
# HISTORY
# 2014-02-04
# o ROBUSTNESS: Now points() for C1C2 passes (modified) argument 'x' to
#   NextMethod() as 'object=x'.
# 2012-09-21 [HB]
# o Now extractDeltaC1C2() makes sure that there are splitters between
#   chromosomes and gaps.
# 2010-10-05 [HB]
# o Added extractPolarDeltaC1C2(), which returns [0,pi] angles and lengths.
# o Now extractDeltaC1C2() returns two types of change-point weights.
# 2010-09-19 [HB]
# o Created.
##############################################################################
